director/redirect.go

```diff
@@ -402,8 +402,13 @@ func registerServeAd(ctx *gin.Context, sType ServerType) {
 		prefix := path.Join("caches", ad.Name)
 		ok, err := VerifyAdvertiseToken(token, prefix)
 		if err != nil {
-			log.Warningln("Failed to verify token:", err)
-			ctx.JSON(400, gin.H{"error": "Authorization token verification failed"})
+			if !ok {
+				log.Warningln("Failed to verify token:", err)
+				ctx.JSON(404, gin.H{"error": "Cache is not admin approved"})
```

Comment:
A few smallish notes:

There don't appear to be any paths in `VerifyAdvertiseToken` that return `true, err`, so all errors from the function will appear like an admin approval error. Instead of trying to parse combinations of `ok` and `err`, you can define a custom error type and check directly for an admin approval error via type assertion like you do elsewhere with `serverCredsErr`.

Also, I think we should be careful about the HTTP status code we reply with. Replying with a `404` here implies to the cache that the director resource it's attempting to `PUT` to couldn't be found. If we're denying its request to advertise because the token doesn't have the correct scopes or the cache isn't approved, we should reply with `403` -- permission denied.

While we're meddling with these sections of code, let's switch the `40X` to the appropriate `http.StatusXXX` methods. For `403`, this is `http.StatusForbidden`